lib/ansible/modules/cloud/azure/azure_rm_sqlserver.py

```diff
@@ -0,0 +1,321 @@
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMServers(AzureRMModuleBase):
+    """Configuration class for an Azure RM SQL Server resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            admin_username=dict(
+                type='str',
+                required=False
+            ),
+            admin_password=dict(
+                type='str',
+                no_log=True,
+                required=False
+            ),
+            version=dict(
+                type='str',
+                required=False
+            ),
+            identity=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+        self.parameters['identity'] = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif key == "tags":
+                self.parameters["tags"] = kwargs[key]
+            elif key == "location":
+                self.parameters["location"] = kwargs[key]
+            elif key == "admin_username":
+                self.parameters["administrator_login"] = kwargs[key]
+            elif key == "admin_password":
+                self.parameters["administrator_login_password"] = kwargs[key]
+            elif key == "version":
+                self.parameters["version"] = kwargs[key]
+            elif key == "identity":
+                self.parameters['identity']["type"] = kwargs[key]
+
+        old_response = None
+        results = dict()
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
```

Comment:
No need for a `try/except` here as azure_rm_common handles a missing source group